wren.go

```diff
@@ -318,6 +344,35 @@ func write(vm *C.WrenVM, text *C.char) {
 	fmt.Fprint(out, C.GoString(text))
 }
 
+//export loadModule
+func loadModule(vm *C.WrenVM, name *C.char) *C.char {
+	var module string = C.GoString(name)
+	var source string
+
+	// Ensure module does not have undesired characters
+	// that can pose thread to remote-code-inclusions
+	if !strings.Contains(module, "..") {
+		// Proceed to load from the configured modules directory only
+		var jvalPtr unsafe.Pointer = C.wrenGetUserData(vm)
+		if jvalPtr != nil {
+			userData := make(map[string]interface{})
+			jval := C.GoString((*C.char)(jvalPtr))
+			if e := json.Unmarshal([]byte(jval), &userData); e == nil {
+				if modulesDir, ok := userData["MODULES_DIR"]; ok {
+					// Precedence (modules_dir/module_name.wren) next (modules_dir/module_name/module.wren)
+					if fdata, e := ioutil.ReadFile(filepath.Join(modulesDir.(string), module) + ".wren"); e == nil {
+						source = string(fdata)
+					} else if fdata, e = ioutil.ReadFile(filepath.Join(modulesDir.(string), module, "module.wren")); e == nil {
+						source = string(fdata)
+					}
```

Comment:
This duplication could be reduced into a method like

```go
func readModule(dir, name string) (string, error) {
    // ...
}
```

We also want to be more careful with error handling. Assuming the precedence is based on file existence, `e == nil` should be made more discrimnatory, since an error can occur if the file doesn't exist, exists with bad permissions, some other low-level failure, etc. [`os.IsNotExist`](https://golang.org/pkg/os/#IsNotExist) is good for checking this case specifically.